mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -261,6 +289,64 @@ def remove_repeated_time_index(ds):  # {{{
     return ds  # }}}
 
 
+def map_variable(variable_name, ds, variable_map):
+    """
+    Find the variable (or list of variables) in dataset ds that map to the
+    mpas_analysis variable given by variable_name.
+
+    variable_map is a dictionary with keys that are variable names used by
+    MPAS-Analysis and values that are lists of possible names for the same
+    variable in the MPAS dycore that produced the data set (which may differ
+    between versions).
+
+    Xylar Asay-Davis
+    12/04/2016
+    """
+    possible_variables = variable_map[variable_name]
+    for var in possible_variables:
+        if isinstance(var, (list, tuple)):
+            allFound = True
+            for subvar in var:
+                if subvar not in ds.data_vars.keys():
+                    allFound = False
+                    break
+            if allFound:
+                return var
+
+        elif var in ds.data_vars.keys():
+            return var
+
+    raise ValueError('Variable {} could not be mapped. None of the '
+                     'possible mapping variables {}\n match any of the '
+                     'variables in {}.'.format(
+                         variable_name, possible_variables,
+                         ds.data_vars.keys()))
+
+
+def rename_variables(ds, variable_map):
+    """
+    Rename all variables in ds based on which are found in variable_map.
+
+    variable_map is a dictionary with keys that are variable names used by
+    MPAS-Analysis and values that are lists of possible names for the same
+    variable in the MPAS dycore that produced the data set (which may differ
+    between versions).
+
+
+    Xylar Asay-Davis
+    12/04/2016
+    """
+    rename_dict = {}
+    for ds_var in ds.data_vars:
+        for map_var in variable_map:
+            rename_list = variable_map[map_var]
+            if ds_var in rename_list:
+                rename_dict[ds_var] = map_var
+                break
```

Comment:
Ah, no, I've done that by mistake and it causes an error in xarray to try to rename a variable to an existing name.

It seems to me that such a conflict should only arise from a bug in the code.  Since the behavior without making any change would be to throw an exception, and the thing I would add would be to throw an exception, can't we just leave it as it is?